prac_08/squaring.py

```diff
@@ -0,0 +1,42 @@
+"""
+CP1404/CP5632 Practical
+Kivy GUI program to square a number
+Roderick Mabo
+Started 09/11/2023
+"""
+
+from kivy.app import App
+from kivy.lang import Builder
+from kivy.core.window import Window
+
+__author__ = 'Roderick Mabo'
+
+
+class SquareNumberApp(App):
+    """ SquareNumberApp is a Kivy App for squaring a number """
+
+    def build(self):
+        """ build the Kivy app from the kv file """
+        Window.size = (200, 100)
+        self.title = "Square Number"
+        self.root = Builder.load_file('squaring.kv')
+        return self.root
+
+    def handle_calculate(self, value):
+        """ handle calculation (could be button press or other call), output result to label widget """
+        try:
+            result = float(value) ** 2
+            self.root.ids.output_label.text = str(result)
+        except ValueError:
+            self.root.ids.output_label.text = '0.0'
```

Comment:
I believe we were meant to set the input_number to blank if a ValueError occured.
What if someone inputted 7c4 and it calculated it as 0.0?
Someone might actually think that's the answer